### PR TITLE
#140: Play sound assigned to step on button click

### LINKED ISSUE #140: Create step - Play sound button
Play sound assigned to step on button click

### PR DESCRIPTION
**Pull request by mikolevy**
- Bind onClick event with handler
- Add playing/stopping sound logic to fragment
- Update Robolectric version to handle Android Binding
- Test

No test for checking is sound played properly because of I still didn't figure out how test sound with Espresso/Robolectric...

Resolves #140

### CODE REVIEW

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java [hunk 1]
```diff
@@ -1,17 +1,111 @@
...
+        }
+    }
+
+    private String retrieveSoundFile() {
+        String soundFileName = assetRepository.get(soundId).getFilename();
+        String fileDir = getActivity().getApplicationContext().getFilesDir().toString();
+        return fileDir + File.separator + soundFileName;
+    }
+
+    private void runBtnAnimation() {
```
Comment (malsolec) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java]: I prefer not to use shortcuts in methods names. I would rather go with runAnimation() and stopAnimation()

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java [hunk 2]
```diff
@@ -1,17 +1,111 @@
...
+        stepSound = (EditText) view.findViewById(R.id.id_et_step_sound);
+        playSoundIcon = (ImageView) view.findViewById(R.id.id_iv_play_step_sound_icon);
+    }
+
+    @Override
+    public void onPlayStopSoundClick(View view) {
+        playStopSound();
+    }
+
+    private void playStopSound() {
```
Comment (malsolec) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java]: Stop?

Comment (malsolec) [reply Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java]: I would rather leave only playSound()

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java [hunk 3]
```diff
@@ -1,17 +1,111 @@
...
+        boolean isNameEmpty = stepSound.getText().toString().isEmpty();
+        if (!isNameEmpty) {
```
Comment (malsolec) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java]: I would rather check if you can perform step  in -> onPlayStopSoundClick()
```
onPlaySoundClick() {
   if(!stepSound.getText().toString().isEmpty()) {
      playStopSound();
   } else {
    showToastMessage(R.string.no_file_to_play_error);
}
}
```

then in playSound only stop and start 

```
playSound() {
if (!mediaPlayer.isPlaying()) {
                 playSound(retrieveSoundFile());
                runBtnAnimation();
            } else {
                 stopSound();
                 stopBtnAnimation();
            }
} 
```
That could help a little with if-ology.
Similar actions are in TaskCreateFragment. Please think about extracting them to separate service.

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java [hunk 4]
```diff
@@ -1,17 +1,111 @@
...
+    }
+
+    private String retrieveSoundFile() {
+        String soundFileName = assetRepository.get(soundId).getFilename();
+        String fileDir = getActivity().getApplicationContext().getFilesDir().toString();
+        return fileDir + File.separator + soundFileName;
+    }
+
+    private void runBtnAnimation() {
+        playSoundIcon.setImageResource(R.drawable.ic_playing_sound_2);
```
Comment (malsolec) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_create/StepCreateFragment.java]: ic_playing_sound_2  : why is it 2?

**Comment by mikolevy** [comment]
Thanks for review @malsolec ! :) Your notices are relevant and I agree with them :) I will prepare separated PR with fragments refactoring (I'll try to extract some components from them) and I'll apply your ideas there

**Comment by malsolec** [comment]
Fine thanks, I will keep that in mind. 🐱

**Pull request merged**